wrappers/python/examples/d500_triggered_calibration.py

```diff
@@ -0,0 +1,213 @@
+    print("Calibration done")
+    # offset to trim:
+    # - 3 bytes for state, progress and result
+    # - 16 bytes for hwm header
+    offset_to_trim = 19
+    depth_calib_after = depth_calib_after[offset_to_trim:]
+    return depth_calib_after
+
+
+def exit_if_calibrations_are_equal():
+    global calib_before, calib_after
+    is_equal_check = (calib_before == calib_after)
+    if is_equal_check:
+        print("Calibrations before and after are the same!!!")
+        exit(1)
+    else:
+        print("Calibration after is different than Calibration before")
+
+
+def compare_results():
+    global fill_rate_before, fill_rate_after
+    print("Comparing Results:")
+    delta_fill_rate_percentage = (fill_rate_after - fill_rate_before) / fill_rate_before
+    if delta_fill_rate_percentage > 0:
+        print("Fill rate improved by " + repr(delta_fill_rate_percentage) + "%")
+    else:
+        print("Fill rate decreased by " + repr(delta_fill_rate_percentage) + "% !!!")
+
+
+def write_after_calib_to_device(calib_filename_after):
+    print("Uploading Calibration to device")
+    global dev
+    curr_path = os.path.dirname(os.path.realpath(__file__))
+    depth_cal_after_path = os.path.join(curr_path, calib_filename_after)
+    new_calib_array = np.fromfile(depth_cal_after_path, dtype=np.uint8)
+    hwm_dev = dev.as_debug_protocol()
+    set_calib_opcode = 0xa6
+    flash_mem_enum = 1
+    depth_calibration_id = 0xb4
+    d500_calib_dynamic = 0
+    header = build_header(new_calib_array)
+    list_header = list(header)
+    new_calib_with_header = np.hstack((header, new_calib_array))
+    cmd = hwm_dev.build_command(opcode=set_calib_opcode,
+                                param1=flash_mem_enum,
+                                param2=depth_calibration_id,
+                                param3=d500_calib_dynamic,
+                                data=new_calib_with_header)
+    ans = hwm_dev.send_and_receive_raw_data(cmd)
+    print("Calibration Uploaded to device")
+    return new_calib_array
+
+
+ctx = rs.context()
+time.sleep(2)
+dev = ctx.query_devices().front()
+
+# Check this device is Safety Camera
+check_d500_camera_found(dev)
+
+# Stream and get Depth Fill Rate
+fill_rate_before = stream_and_get_fill_rate()
+
+# Get existing depth calibration
+calib_before = get_depth_calib_from_device()
+
+# Save existing calibration to file
+save_calib_to_file(calib_before, "depth_calib_before.bin")
+
+# Trigger Calibration - log progress
+calib_after = run_calibration()
+
+# Save new depth calibration to file
+save_calib_to_file(calib_after, "depth_calib_after.bin")
+
+# Stopping the script if the calibration received from the calibration operation
+# is the same as before
+exit_if_calibrations_are_equal()
+
+# Set calibration from file - needed until the flash writing is implemented in HKR
+write_after_calib_to_device("depth_calib_after.bin")
```

Comment:
Do we need to write to file and read from it later? Can't we pass `calib_after` as a parameter?

Reply:
I don't see the need for it here.
Please see the example flow again and tell me if it is needed and for what.

Reply:
As I understand the flow we
- save to file the original table
- calibrate
- save to file the updated table
- read from file the updated table
- sending to the device the table we have read from file

My question is why do we need to save to files?

Reply:
saving the files helps to see the difference between before and after.
Moreover, if the new calibration seems not better, we can use the file saved before the calibration to write it back to the device. So, I think it is better to save these files.
BTW, now that the calibration is written by the FW when the process succeeds, the function "write_after_calib_to_device" is not needed anymore - so I am removing it.